indico/web/client/js/utils/date.js

```diff
@@ -55,3 +55,31 @@ export function dayRange(start, end, step = 1) {
     }
     return result;
 }
+
+export function createDt(date, time) {
+    const momentDate = moment(date, 'YYYY-MM-DD');
+    const momentTime = moment(time, 'HH:mm');
+    if (!momentDate.isValid() || !momentTime.isValid()) {
+        return null;
+    }
+
+    return moment([...momentDate.toArray().splice(0, 3), ...momentTime.toArray().splice(3)]);
+}
+
+function isBookingStartValid(dt, isAdminOverrideEnabled = false, granularity = 'minute') {
+    if (!dt || !dt.isValid()) {
+        return false;
+    } else if (isAdminOverrideEnabled) {
+        return true;
+    }
+    const gracePeriod = granularity === 'day' ? moment() : moment().subtract(1, 'hour');
```

Comment:
:thinking: I wonder whether this grace period will be enough. I'm more inclined to be more flexible. Anyway we can wait if we get any complaint and modify it afterwards.

Reply:
I think that is a good compromise. Anyway, we will be able to easily change it later on if there is a need.